importlib_resources/readers.py

```diff
@@ -83,12 +83,28 @@ def is_file(self):
         return False
 
     def joinpath(self, *descendants):
-        try:
-            return super().joinpath(*descendants)
```

Comment:
I'm sad to lose the re-use of the generalized behavior, though I'm not confident there's any other way. I'd like to review the generalized behavior to see if there's yet any opportunity for re-use.

Reply:
I took a look at the implementation for `Traversable.joinpath` and I do think maybe it could be made suitable. I think the problem with the `Traversable.joinpath` is that when it calls `iterdir`, it uses `unique_everseen(key=name)` on the results, which would cause only one directory to resolve even when there are many. Probably instead, `MultiplexedPath.iterdir` should return another `MultiplexedPath` when one of the children matches in more than one of `self._paths`.

Reply:
In #278, I believe I've been able to re-use Traversable.joinpath by correcting the behavior in MultiplexedPath.iterdir to honor multiple subdirectories.